.github/workflows/check.yml

```diff
@@ -25,16 +25,26 @@ jobs:
         - pypy3
         - pypy2
     steps:
+    - name: Setup graphviz
+      uses: ts-graphviz/setup-graphviz@v1
+    - name: Setup python for tox
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install tox
+      run: python -m pip install tox
     - name: Setup python for test ${{ matrix.py }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.py }}
     - uses: actions/checkout@v2
-    - name: Setup graphviz
-      uses: ts-graphviz/setup-graphviz@v1
-    - name: Install dev requirements
-      run: pip install -r dev-requirements.txt
-    - name: Install project
-      run: pip install .[graphviz]
+    - name: Pick tox environment to run
+      run: |
+        import subprocess; import json
+        major, minor, impl = json.loads(subprocess.check_output(["python", "-c", "import json; import sys; import platform; print(json.dumps([sys.version_info[0], sys.version_info[1], platform.python_implementation()]));"], universal_newlines=True))
```

Comment:
Didn't understand why another python process needs to be started. If `sys.version_info` and `platform.python_implementation` are called in the same process, won't it return the same results?

Reply:
I mean the following 

```
- name: Pick tox environment to run
  shell: python
  run: |
    import sys; import platform
    major, minor = sys.version_info[:2]
    impl  = platform.python_implementation()
    print('::set-env name=TOXENV::' + ("py" if impl == "CPython" else "pypy") + ("{}{}".format(major, minor) if impl == "CPython" else ("3" if major == 3 else "")))
```

Reply:
```
py -c 'import sys; import platform; print(platform.python_implementation());print(sys.version_info)'                                                                                                                                          CPython
sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)
```

```
pypy3  -c 'import sys; import platform; print(platform.python_implementation());print(sys.version_info)'                                                                                                                                      PyPy
(major=3, minor=6, micro=9, releaselevel='final', serial=0)
```

They return different things.

Reply:
In the run step `python -c` is called. Wouldn't the python shell itself be of the same implementation and version as `python -c` subprocess created from it?

Reply:
There's no python shell and python -c. There's only one process called there. shell:python transaltes to python -c.

Reply:
Ah you're right.